ceph/rados/mgr_workflows.py

```diff
@@ -0,0 +1,274 @@
+
+    def remove_mgr_service(self, host) -> bool:
+        """
+        Method to remove the mgr service from the given host
+
+        Args:
+            host: name of the host from where mgr needs to be removed
+
+        returns:
+            Pass -> True, Fail -> false
+        """
+        mgr_daemon_list = self.get_mgr_daemon_list()
+        new_mgr_daemon = ""
+        for node in mgr_daemon_list:
+            if host.hostname in node:
+                new_mgr_daemon = node.strip()
+                break
+
+        host_name = host.hostname
+        if not self.check_mgr_exists_on_host(host=host_name):
+            log.info(f"MGR daemon not present on the host {host_name}. Returning Pass")
+            return True
+
+        # setting mgr daemon unmanaged to true
+        unmanaged_cmd_true = " ceph orch apply mgr  --unmanaged=true"
+        self.client.exec_command(sudo=True, cmd=unmanaged_cmd_true)
+
+        cmd = f"ceph orch daemon rm mgr.{new_mgr_daemon} --force"
+        try:
+            self.client.exec_command(sudo=True, cmd=cmd)
+        except Exception as err:
+            log.debug(f"Hit issue during command execution : {err}")
+            log.debug("proceeding to check if the command execution was successful")
+
+        mgr_exists = True
+        endtime = datetime.datetime.now() + datetime.timedelta(seconds=180)
+        while datetime.datetime.now() < endtime:
+            if not self.check_mgr_exists_on_host(host=host_name):
+                log.info(f"Mgr daemon is removed on the host {host_name}.")
+                mgr_exists = False
+                break
+            log.info(
+                f"Mgr daemon still present on the host {host_name}. sleeping for 5 seconds and checking again"
+            )
+            time.sleep(5)
+        if mgr_exists:
+            log.error(
+                f"Mgr daemon still present on the host {host_name}. Failed to remove"
+            )
+            return False
+
+        # Removing mgr label from host if present
+        labels = self.mon_obj.get_host_labels(host=host_name)
+        if "mgr" in labels:
+            cmd = f"ceph orch host label rm {host_name} mgr"
+            self.client.exec_command(sudo=True, cmd=cmd)
+            log.debug(f"Removed mgr label from host : {host_name}")
+            # setting mgr daemon unmanaged to true
+
+        unmanaged_cmd_false = " ceph orch apply mgr  --unmanaged=false"
+        self.client.exec_command(sudo=True, cmd=unmanaged_cmd_false)
+        log.debug(f"Mgr successfully removed from host : {host_name}")
+        return True
+
+    def mgr_scale_up_down(self, mgr_node_list):
+        mgr_daemons = self.get_mgr_daemon_list()
+        mgr_count = len(mgr_daemons)
+        new_mgr_count = len(mgr_node_list)
+
+        additional_mgr_count = new_mgr_count - mgr_count
+        if mgr_count == new_mgr_count:
+            log.info(
+                " The request count is same as the current mgr count.Not performing any scale operation"
```

Comment:
What if the count is same but items in the list are different?

Reply:
Very valid comment. I  will modify the code to check the mar daemons instead of count.

Reply:
The following scenarios need to be covered:
1. The equal number of requested mgr lists is totally different from the existing mgr 
2. Scale up the new mgr list that is different from the existing mgr's.
3. Scale down the new mgr list that is different from the existing mgr's.

This scenario will be covered in the Jira task - https://issues.redhat.com/browse/RHCEPHQE-14793

Reply:
The logic implemented.